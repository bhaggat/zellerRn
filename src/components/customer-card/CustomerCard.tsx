import React, {useMemo} from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import {Customer} from '../../services/graphql/customers';
import {useNavigation, NavigationProp} from '@react-navigation/native';

export type RootStackParamList = {
  Details: {customer: Customer};
};

type Navigation = NavigationProp<RootStackParamList, 'Details'>;

type CustomerCardProps = {
  customer: Customer;
};

const CustomerCard = ({customer}: CustomerCardProps) => {
  const {name, role} = customer;
  const firstLetter = useMemo(() => name.charAt(0).toUpperCase(), [name]);
  const navigation = useNavigation<Navigation>();

  const onCustomerPress = () => {
    navigation.navigate('Details', {customer});
  };

  return (
    <TouchableOpacity onPress={onCustomerPress} style={styles.card}>
      <View style={styles.avatar}>
        <Text style={styles.avatarText}>{firstLetter}</Text>
      </View>
      <View>
        <Text style={styles.name} testID="customer-name">
          {name}
        </Text>
        <Text style={styles.role} testID="customer-role">
          {role}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  card: {
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 8,
    gap: 10,
  },
  avatar: {
    width: 40,
    height: 40,
    backgroundColor: '#D1E4FC',
    borderRadius: 4,
    justifyContent: 'center',
    alignItems: 'center',
  },
  avatarText: {
    fontSize: 20,
    color: '#007BFF',
  },
  name: {
    fontSize: 16,
    fontWeight: '500',
  },
  role: {
    fontSize: 14,
    color: '#6B7280',
    textTransform: 'capitalize',
  },
});

export default CustomerCard;
