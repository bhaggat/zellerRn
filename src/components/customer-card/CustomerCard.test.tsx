import {render} from '@testing-library/react-native';
import CustomerCard from './CustomerCard';

describe('CustomerCard Component', () => {
  const customer = {
    id: '1',
    name: 'Dhruv Bhagat',
    email: 'dhruvbhagat98@gmail.com',
    role: 'Admin',
  };
  describe('Snapshot', () => {
    test('matches', () => {
      const tree = render(<CustomerCard customer={customer} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('Render', () => {
    test('Component render based on props', () => {
      const {getByTestId} = render(<CustomerCard customer={customer} />);
      expect(getByTestId('customer-name').props.children).toBe(customer.name);
      expect(getByTestId('customer-role').props.children).toBe(customer.role);
    });
  });
});
