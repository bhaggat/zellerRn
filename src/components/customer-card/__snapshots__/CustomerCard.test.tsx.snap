// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerCard Component Snapshot matches 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderRadius": 8,
      "flexDirection": "row",
      "gap": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#D1E4FC",
        "borderRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      }
    }
  >
    <Text
      style={
        {
          "color": "#007BFF",
          "fontSize": 20,
        }
      }
    >
      D
    </Text>
  </View>
  <View>
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
      testID="customer-name"
    >
      Dhruv Bhagat
    </Text>
    <Text
      style={
        {
          "color": "#6B7280",
          "fontSize": 14,
          "textTransform": "capitalize",
        }
      }
      testID="customer-role"
    >
      Admin
    </Text>
  </View>
</View>
`;
